core/imageroot/usr/local/agent/pypkg/cluster/grants.py

```diff
@@ -99,3 +99,36 @@ def alter_user(rdb, user, revoke, role, on_clause):
 
     pipe.execute()
 
+def refresh_permissions(rdb):
+    """Scan the stored authorizations and re-apply permissions to all modules"""
+    for kauth_module in rdb.scan_iter('cluster/authorizations/module/*'):
+        kmid = kauth_module.removeprefix('cluster/authorizations/module/')
+        authorizations = rdb.smembers(kauth_module) or []
+        mnode_id = rdb.hget('cluster/module_node', kmid)
+        add_module_permissions(rdb, kmid, authorizations, mnode_id)
+
+def add_module_permissions(rdb, module_id, authorizations, node_id):
+    """Parse authorizations and grant permissions to module_id"""
+    for authz in authorizations:
+        if '*' in authz or '[' in authz or '?' in authz:
+            continue # Redis glob patterns are forbidden
```

Comment:
Now it's better.
But since there is no strong validation for `authz` why not removing the check?

Reply:
I want to keep the check and implement a simple validation rule.

Reply:
:shrug: